### ISSUE TITLE
`triton inst create ...` is broken if "images.json" cache file is missing

### ISSUE THREAD

**Issue by KrisShannon** [joyent/node-triton#93]
```shell
triton --version
```
triton [4.5.1](https://github.com/joyent/node-triton/commits/4.5.1)
```shell
triton instance create -w "$IMG_UUID" "$PKG_UUID"
```
```javascript
/opt/local/lib/node_modules/triton/lib/tritonapi.js:357
                    for (var i = 0; i < images.length; i++) {
                                              ^
```
TypeError: Cannot read property 'length' of undefined
    at [/opt/local/lib/node_modules/triton/lib/tritonapi.js:357:47](https://github.com/joyent/node-triton/blob/4.5.1/lib/tritonapi.js#L357)
    at [/opt/local/lib/node_modules/triton/lib/tritonapi.js:219:20](https://github.com/joyent/node-triton/blob/4.5.1/lib/tritonapi.js#L219)
    at [fs.js:208:20](https://github.com/nodejs/node-v0.x-archive/blob/v0.10.40/lib/fs.js#L208)
    at [Object.oncomplete (fs.js:108:15)](https://github.com/nodejs/node-v0.x-archive/blob/v0.10.40/lib/fs.js#L108)
```

**Comment by KrisShannon** [comment]
It looks like it's because [_cacheGetJson](https://github.com/joyent/node-triton/blob/4.5.1/lib/tritonapi.js#L208-L242) returns null if the [cache file does not exist](https://github.com/joyent/node-triton/blob/4.5.1/lib/tritonapi.js#L217-L219).

The [call to _cacheGetJson in getImage](https://github.com/joyent/node-triton/blob/4.5.1/lib/tritonapi.js#L352-L364) is assuming the return must be an array.

**Comment by trentm** [comment]
Thanks, again! Your fix will be published in version 4.5.2 shortly.
